register/models.py

```diff
@@ -76,9 +76,13 @@ def invite(self, request):
         self.status = 'I'
         self.save()
 
-    def confirm(self):
-        if self.status != 'I':
+    def is_confirmed(self):
+        return self.status == 'C'
+
+    def confirm(self, cancellation_url):
+        if self.status != 'I' and self.status != 'C':
             raise ValidationError('Application hasn\'t been invited yet')
+        self._send_confirmation_ack(cancellation_url)
```

Comment:
Send only once, only if accepted for the first time.  Same for the status, only update application if it's not confirmed already (no need to do unnecessary updates to the DB)